### PR TITLE
Allow pinning of coverage reporter version

### PR DESCRIPTION
**Pull request by littleforest**
This add a `coverage-reporter-version` parameter that allows users to specify a version of [coverage-reporter](https://github.com/coverallsapp/coverage-reporter) to use. It will default to the latest version if not specified. This currently only works for Linux and Windows until versioning can be added to the Coveralls Homebrew tap.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -79,27 +83,47 @@ runs:
         brew tap coverallsapp/coveralls --quiet
         brew install coveralls --quiet
 
+    - name: Report coverage-reporter-version information for macOS
+      if: ${{ runner.os == 'macOS' && inputs.coverage-reporter-version != 'latest' }}
+      shell: bash
+      run: echo "The coverage-reporter-version parameter is not available on macOS"
```
Comment (mike-burns) [inline action.yml:89]: Do we want to exit early here? Do something with stderr? I don't really know our options.

Naively:

```suggestion
      run: |
        echo "The coverage-reporter-version parameter is not available on macOS" >&2
        exit 1
```

**Review by mike-burns — COMMENTED** [review]
Adding "Powershell expert" to your skills list.

File: action.yml (hunk 1 above)
Comment (littleforest) [reply action.yml]: Screenshot 2024-05-06 at 11 30 34 AM

**Review by afinetooth — APPROVED** [review]
Looks great. Thank you. 🙏

One request, not related to this PR:

We need to offer a parallel configuration option in our CircleCI Orb. 

Will you please create a new ticket like this one for the Orb? 

Ticket should live in Coveralls project, even though PR will live in the Orb repo. 🙏

**Pull request merged**